src/zoom.ts

```diff
@@ -1,257 +1,213 @@
-				);
-				break;
-			}
-
-			participants.push({
-				name: event.payload.object.participant.user_name,
-				id: event.payload.object.participant.user_id,
-			});
-
-			if (
-				ZOOM_TIME_THRESHOLD &&
-				ZOOM_TIME_ANNOUNCEMENT_CHANNEL &&
-				participants.length === ZOOM_TIME_THRESHOLD &&
-				Date.now() - lastZoomTime >= ZOOM_TIME_DEBOUNCE_HOURS * 60 * 60 * 10000
-			) {
-				lastZoomTime = Date.now();
-				const participantsStr =
-					participants.length === 1
-						? participants[0].name
-						: `${participants
-								.slice(0, -1)
-								.map((p) => p.name)
-								.join(',')}, and ${participants[participants.length - 1].name}`;
-
-				await Promise.all(
-					discord.guilds.cache.map(async (guild) => {
-						const channel = guild.channels.cache.find(
-							(ch) =>
-								ch.name.toLowerCase() ===
-								ZOOM_TIME_ANNOUNCEMENT_CHANNEL?.toLowerCase(),
-						) as TextChannel | undefined;
-
-						if (!channel) return;
-
-						await channel.send(formatMessage(guild)`
-🚨 Paging everybody, ${participantsStr} are starting a call, it's ${`%Zoom Time`}! 🚨
-						`);
-					}),
-				);
-			}
-			break;
-		}
-		case 'meeting.participant_left': {
-			zoomInfo = await getZoomInfo();
-			if (!zoomInfo.active) {
-				console.log(
-					'WARNING(webhook): Zoom participant left while meeting was inactive. Ignoring.',
-				);
-				break;
-			}
-
-			participants = participants.filter(
-				(participant) => participant.id !== event.payload.object.participant.user_id,
-			);
-
-			break;
-		}
-		default: {
-			throw new ResponseError(`Unknown Webhook Event!: ${(event as ZoomEvent).event}`, 400);
-		}
-	}
-	await updateDiscordStatusFromZoom(discord, zoomInfo);
+const zoomId = (
+  event: ZoomParticipantJoinedEvent | ZoomParticipantLeftEvent
+) => {
+  const participant = event.payload.object.participant;
+  return (
+    participant.id || `${event.payload.object.uuid}.${participant.user_id}`
+  );
+};
+
+export async function processWebhookEvent(event: ZoomEvent) {
+  console.log(`Processing Zoom Webhook Event: ${event.event}`);
+  if (!PRODUCTION) console.log(JSON.stringify(event, null, 2));
+
+  const { id: meetingId } = event.payload.object;
+
+  // We toString both IDs because they're numerical and I don't want funny type errors.
+  if (!meetingId || meetingId.toString() !== ZOOM_MEETING_ID.toString()) {
+    console.log(`Webhook is for a different meeting (${meetingId}), ignoring.`);
+    return;
+  }
+
+  await updateZoomStatus();
+
+  switch (event.event) {
+    // We don't want to trust Zoom's webhooks for synchronization reasons, so we re-fetch the status
+    case "meeting.started":
+    case "meeting.ended": {
+      break;
+    }
+    case "meeting.participant_joined": {
+      const participant = event.payload.object.participant;
+      dispatch(
+        userJoined({
+          zoomId: zoomId(event),
+          name: participant.user_name,
+          temporary: !Boolean(participant.id),
```

Comment:
Nit: don't use `Boolean`, especially if you're inverting it anyways. Just use `!participant.id`